### PR TITLE
Feature/4 apply tailwind theme system to appwide

### PR DESCRIPTION
**Pull request by mjeylanii**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -20,7 +20,6 @@ next-env.d.ts
 # expo
 .expo/
 dist/
-expo-env.d.ts
```
Comment (zxcvbnmmohd) [inline .gitignore:23]: why'd ya remove this?

File: apps/expo/src/app/friends-mode.tsx [hunk 2]
```diff
@@ -9,11 +9,11 @@ export default function Page(): ReactNode {
 	const isPresented = router.canGoBack()
 
 	return (
-		<SafeAreaView className="bg-[#1F104A]">
+		<SafeAreaView>
 			<Stack.Screen
 				options={{ title: "Friends Mode", presentation: "modal" }}
 			/>
-			<View className="flex-col items-center  gap-3">
+			<View className="flex-col items-center gap-3">
```
Comment (zxcvbnmmohd) [inline apps/expo/src/app/friends-mode.tsx]: don't forget to apply the same styling as above 
const styles = {
...
}

File: apps/expo/src/app/index.tsx [hunk 3]
```diff
@@ -54,7 +59,9 @@ export default function Page(): ReactNode {
 								scale: { type: "spring", delay: 350 },
 							}}
 						>
-							<Text className="text-8xl font-bold text-violet-700">
+							<Text
+								className={styles.logo + " " + "text-primary"}
```
Comment (zxcvbnmmohd) [inline apps/expo/src/app/index.tsx]: try to apply the way too avoid adding  in the class name, just mane a new option, it's calm

File: apps/expo/src/app/result/index.tsx [hunk 4]
```diff
@@ -20,6 +20,20 @@ type StatItemProps = {
 }
 
 export default function Page(): ReactNode {
+	const styles = {
+		body: "flex size-full flex-col items-center justify-between",
+		headerContainer: "flex w-full flex-row items-end justify-between p-4",
+		headerText: "ml-6 text-2xl font-bold",
+		statsContainer:
+			"mt-96 flex w-full flex-row items-center justify-center px-10",
+		bottomBtnsContainer: "flex h-14 flex-row gap-3 px-12",
```
Comment (zxcvbnmmohd) [inline apps/expo/src/app/result/index.tsx]: don't abbreviate words

File: apps/expo/src/app/result/index.tsx [hunk 5]
```diff
@@ -20,6 +20,20 @@ type StatItemProps = {
...
+		checkAnswersBtn: "absolute bottom-4 mx-10 rounded-xl",
+		shareButton:
+			"flex size-16  items-center justify-center rounded-3xl border-4 border-gray-300",
+		animationContainer:
+			"absolute -z-10 flex size-full items-center  justify-start",
+		animationContainerBg: "absolute top-40 size-96 rounded-3xl bg-primary",
```
Comment (zxcvbnmmohd) [inline apps/expo/src/app/result/index.tsx]: here too

File: apps/expo/src/app/result/index.tsx [hunk 6]
```diff
@@ -97,16 +111,19 @@ export default function Page(): ReactNode {
...
-					<Link href="/">
-						<Button
-							label="Done"
-							onPress={() => {
-								resetSoloSession()
-							}}
-						/>
-					</Link>
-					<Pressable className="flex size-16  items-center justify-center  rounded-3xl border-4 border-gray-300  ">
+				<View className={"flex h-14 flex-row gap-3 px-12"}>
```
Comment (zxcvbnmmohd) [inline apps/expo/src/app/result/index.tsx]: forgot to update this one

File: apps/expo/src/components/Button.tsx [hunk 7]
```diff
@@ -6,23 +6,71 @@ import { Text, TouchableOpacity } from "react-native"
 import { tv } from "tailwind-variants"
 
 type Props = {
+	className?: string
```
Comment (zxcvbnmmohd) [inline apps/expo/src/components/Button.tsx:10]: put optional properties below required ones

File: apps/expo/src/components/Button.tsx [hunk 8]
```diff
@@ -6,23 +6,71 @@ import { Text, TouchableOpacity } from "react-native"
...
 	label: string
 	outline?: boolean
 	onPress?: ((event: GestureResponderEvent) => void) | undefined
-	color?: "primary" | "secondary"
-	size?: "sm" | "md" | "lg"
+	size?: "sm" | "md" | "lg" | "xl" | "2xl" | "3xl"
 }
 
 const Button = (props: Props): React.ReactNode => {
+	const buttonVariant = tv({
```
Comment (zxcvbnmmohd) [inline apps/expo/src/components/Button.tsx]: this really cooooool ngl

File: apps/expo/src/components/Button.tsx [hunk 9]
```diff
@@ -6,23 +6,71 @@ import { Text, TouchableOpacity } from "react-native"
...
+				secondary: "text-white",
+				tertiary: "text-white",
+				accent: "text-white",
+				outline: "text-primary",
+				white: "text-black",
+			},
+		},
+	})
+	const buttonStyle = buttonVariant({ color: props.color, size: props.size })
+	const textStyle = textVariant({ size: props.size, color: props.color })
```
Comment (zxcvbnmmohd) [inline apps/expo/src/components/Button.tsx]: can you do a convert these to variants? lke
`const styles = {
text: textVariant({...}),
button:  buttonVariant({...}), etc...
}`

File: apps/expo/src/components/gameplay/QuestionHeader.tsx [hunk 10]
```diff
@@ -19,42 +19,23 @@ const QuestionHeader = ({
 }): ReactNode => {
 	const { theme } = useTheme()
```
Comment (zxcvbnmmohd) [inline apps/expo/src/components/gameplay/QuestionHeader.tsx]: we still using this?

**Review by zxcvbnmmohd — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (mjeylanii) [reply .gitignore:23]: I didn't touch that

**Review by zxcvbnmmohd — APPROVED** [review]

**Pull request merged**